### ISSUE TITLE
Feature request: Garbage cleanup only after amount of time

### ISSUE THREAD

**Issue by tylercollier** [dereuromark/cakephp-queue#20]
Awesome tool! We are using it in our project.

In my project, I don't want to clean up old jobs for about a week. That way, if there is a problem, I'll have a little history to look back on.

I am accomplishing this today by setting:

```
$config['Queue']['gcprop'] = 0;
```

and then running a separate process weekly to clear the queue of old jobs.

But it would be nice to have this built into cakephp-queue. How about a setting like:

```
$config['Queue']['gcminage'] = 604800; // one week in seconds
```

`gcminage` stands for 'garbage collection minimum age'. Meaning, it won't consider a job as garbage unless it is a certain age, thus it won't clean it out.

This setting would be orthogonal to 'gcprop'. Regardless of the chance of garbage collection, when it runs, it will respect the age of the job and only delete it if appropriate.

What do you think?

**Comment by tylercollier** [comment]
Well go figure, this already exists. It's the Queue.cleanuptimeout property.
